src/dashboard/src/components/filesystem_ajax/views.py

```diff
@@ -831,6 +832,23 @@ def _copy_files_to_arrange(
     return to_add
 
 
+def _save_sip_arranges(arranges):
+    """Bulk create a list of SIPArrange model instances.
+
+    If some of the SIPArrange instances already exist, bulk creation
+    will fail and this will revert back to saving each instance
+    individually ignoring the existing ones.
+    """
+    try:
+        models.SIPArrange.objects.bulk_create(arranges, BULK_CREATE_BATCH_SIZE)
+    except IntegrityError:
+        for arrange in arranges:
+            try:
+                arrange.save()
+            except IntegrityError:
+                continue
```

Comment:
> Would it be correct to say that this is only needed because we don't have Django v2.2's `ignore_conflicts`?

That's correct. Cole pointed me to this strategy for `1.8` back when I first noticed that `bulk_create` could raise the `IntegrityError`s.

> Btw we have a `Directory.create_many` which I like because of the separation of duties, do you think we should create a similar `SIPArrange.create_many`?

Ah, nice! :+1: